### ISSUE TITLE
[Makefile] Impossible to setup on Windows

### ISSUE THREAD

**Issue by AlexV525** [flutter/website#8272]
### Describe the problem

https://github.com/flutter/website/blob/5494bf1ac41f10fe0fb236c66677c1b5a5c8c3dc/Makefile#L172-L173

Since most of the commands start with `clean`, there is no way to run `rm` on Windows command lines.

```dart
C:\ProgramData\chocolatey\bin\make.exe -f X:\Coding\Garages\GitHub\flutter-website\Makefile -C X:\Coding\Garages\GitHub\flutter-website build
process_begin: CreateProcess(NULL, git rev-parse --short HEAD, ...) failed.
X:\Coding\Garages\GitHub\flutter-website\Makefile:12: pipe: No error
make: Entering directory 'X:/Coding/Garages/GitHub/flutter-website'
make clean
make[1]: Entering directory 'X:/Coding/Garages/GitHub/flutter-website'
rm -rf _site .jekyll* src/.jekyll* *.log tmp .dart_tool
make[1]: Leaving directory 'X:/Coding/Garages/GitHub/flutter-website'
process_begin: CreateProcess(NULL, git rev-parse --short HEAD, ...) failed.
Makefile:12: pipe: No error
process_begin: CreateProcess(NULL, rm -rf _site .jekyll* src/.jekyll* *.log tmp
.dart_tool, ...) failed.
make (e=2): 系统找不到指定的文件。
make[1]: *** [Makefile:153: clean] Error 2
make: *** [X:\Coding\Garages\GitHub\flutter-website\Makefile:135: build] Error 2
make: Leaving directory 'X:/Coding/Garages/GitHub/flutter-website'

Process finished with exit code 2

```

**Comment by AlexV525** [comment]
Also, copying Flutter from the host will cause platform issues that unable to start any Flutter commands.

**Comment by atsansone** [comment]
@AlexV525 : Your second issue is a Windows Git issue.

Check out this [StackOverflow post](https://stackoverflow.com/questions/29045140/env-bash-r-no-such-file-or-directory) that explains the issue.

TL;DR: Git translated UNIX line breaks to Windows line breaks on `git clone`. `bash` has no idea what to do with those line breaks.

```terminal
$ git config --global core.autocrlf input
```

That ensured that all _writes_ to git and GitHub use LF only. 

```terminal
$ git config --global core.autocrlf false
```

This ensures that all _reads_ to git and GitHub use LF only. 

A deeper dive can be found on [this GitHub page](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings).

**Comment by atsansone** [comment]
Closing as duplicate of https://github.com/flutter/website/issues/6443.

**Comment by AlexV525** [comment]
> @AlexV525 : Your second issue is a Windows Git issue.

It's unfriendly to ask users to reconfigure their global configurations if they're not unnecessary, like Git, if the step can be done to solve them externally.
